### PR TITLE
Enable checks for encrypted flash in OTA

### PR DESCRIPTION
**Pull request by rojer**
Even if firmware is compiled without CONFIG_FLASH_ENCRYPTION_ENABLED
Rayionale: CONFIG_FLASH_ENCRYPTION_ENABLED controls whether boot loader
generates keys for encryption or not, but flash encryption can be
configured externally. With this change, it's possible to have boot
loader not generate keys but still have encryption working.

### CODE REVIEW

File: components/app_update/esp_ota_ops.c [hunk 1]
```diff
@@ -166,7 +163,7 @@ esp_err_t esp_ota_write(esp_ota_handle_t handle, const void *data, size_t size)
                         return ESP_OK; /* nothing to write yet, just filling buffer */
                     }
                     /* write 16 byte to partition */
-                    ret = esp_partition_write(&it->part, it->wrote_size, it->partial_data, 16);
+                    ret = esp_partition_write(it->part, it->wrote_size, it->partial_data, 16);
```
Comment (edmund-huber) [inline components/app_update/esp_ota_ops.c]: `&it->part` changed to `it->part`. Judging from the signature of esp_partition_write this wasn't compiling properly..

Comment (rojer) [inline components/app_update/esp_ota_ops.c]: yep. a drive-by fix. it generated a compilation warning that must've been overlooked. it could not have possibly worked.

**Comment by igrr** [comment]
Rebased and merged in a845f21, thanks!
